### PR TITLE
Update documentation for CLI v3 launch

### PR DESCRIPTION
**Pull request by dse0**
New content:
- Developer reference (most pages have small updates)
- Developer reference -> Code review & continuous integration
- Developer reference -> CLI reference -> v3 -> Migration guide

### CODE REVIEW

**Comment by vndrewlee** [comment]
i think the only major changes on this branch are the folders for v2 and the build page? is that right?

**Comment by dse0** [comment]
> i think the only major changes on this branch are the folders for v2 and the build page? is that right?

The CLI quickstart and overview pages were rewritten:
https://docs.hashboard.com/docs/data-ops/quickstart
https://docs.hashboard.com/docs/data-ops/managing-resources-with-code

File: pages/docs/data-ops/managing-resources-with-code.mdx [hunk 1]
```diff
@@ -48,6 +87,13 @@ the file structure of your project.
...
+
+Builds and [Project Drafts](/docs/project-management/drafts) serve similar purposes -- they both can be used to stage, test, and apply changes. Here is how they differ:
 
-To sync your local config files with edits made to resources via the UI, use the [`pull`](/docs/data-ops/cli/pull) command.
+|                                            | Builds                                          | Project drafts   |
+| ------------------------------------------ | ----------------------------------------------- | ---------------- |
+| Created from...                            | The CLI or git integration                      | The Hashboard UI |
+| Can be edited?                             | No -- the changes within a build are immutable. | Yes              |
+| Supports environment variable substitution | Yes                                             | No               |
+| Can be deployed to multiple projects       | Yes                                             | No               |
```
Comment (annanixon) [inline pages/docs/data-ops/managing-resources-with-code.mdx:99]: Hmm idk if I agree with the last statement, the resource files can be deployed to multiple projects but the build itself is still tied to a single project just like a draft

File: pages/docs/data-ops.mdx [hunk 2]
```diff
@@ -39,14 +39,14 @@ Hashboard allows you to represent your resources as configuration files, which c
   <Card
     icon={<Terminal {...iconProps} />}
     title="Explore Hashboard commands"
-    href="/docs/data-ops/cli"
+    href="/docs/data-ops/cli/v3"
```
Comment (annanixon) [inline pages/docs/data-ops.mdx:42]: this link seems to be broken in the vercel preview

File: pages/docs/data-ops/quickstart.mdx
Comment (annanixon) [inline pages/docs/data-ops/quickstart.mdx:1]: we should add hb init here

File: pages/docs/data-ops.mdx [hunk 3]
```diff
@@ -39,14 +39,14 @@ Hashboard allows you to represent your resources as configuration files, which c
...
   >
     Master Hashboard's CLI with these essential commands
   </Card>
   <Card
     icon={<CheckCircle {...iconProps} />}
     title="Validate and preview updates"
```
Comment (annanixon) [inline pages/docs/data-ops.mdx:48]: The title/description of this card no longer really matches the content in the link, think we should update the build page to talk about previewing a bit more or update this copy

File: pages/docs/data-ops.mdx
Comment (annanixon) [inline pages/docs/data-ops.mdx:1]: The code review card link is broken in the preview

Comment (annanixon) [reply pages/docs/data-ops.mdx:1]: actually since code review and CI/CD are in a single link maybe we should update the code review card here to point to the dbt integration docs?

File: pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx
Comment (annanixon) [inline pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx:1]: should this be at the same level as v2/v3 instead of nested under v3? I understand why this makes sense semantically but might be worth bumping it up a level right now just for visibility

File: pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+```bash
+$ ls
+models		explorations			dashboards
+
+$ hb build dashboards/*  # Creates a build scoped just to your dashboards
+```
+
+Note that builds can still delete downstream content.  For example, if you update a model and remove one of its attributes, Explorations which depend on that attribute may be slated for removal, even if they are not in the build.
+
+Deployment of all code-controlled resources is still available by supplying a `--full_rebuild` flag:
```
Comment (annanixon) [inline pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx:30]: Would reword to `Rebuilding all code-controlled resources in the project is still ...`

File: pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+```bash
+$ hb build . --full-rebuild  # Will delete any missing Hashboard resources that
+                             # were previously deployed with code
+```
+
+### Project root file
+
+The v2 CLI was sensitive to where you ran it (your current working directory) relative to your yml files. If you didn’t run `hb` from the right location, your resource IDs might unexpectedly change.
+
+The v3 CLI introduces a `hb init` command, which creates a `.hbproject` file in your filesystem. This file is to be placed at the root of your Hashboard project and ensures that your resource identities are stable, regardless of where you are running your `hb build` command.
```
Comment (annanixon) [inline pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx:41]: Should we specify the caveat that build must always be run from a subdirectory?

File: pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+2. Run `hb init` at the root of your Hashboard project files, specifying `--dbt-root=...` if you have dbt-based models. Commit the resulting `.hbproject` file into your git repository.
+3. In your scripts / CI actions, migrate commands as follows:
+
+
+
+    | **Previous command** | **New command** |
+    | --- | --- |
+    | `hb preview (filepath)` | `hb build (filepath)` <br/><br/> Include the `--full_rebuild` flag if you want to delete missing resources. |
+    | `hb deploy (filepath)`  | If the build was already created for previewing: `hb build apply`<br/><br/>If no build has been created yet:<br/>`hb build (filepaths) && hb build apply (--no-confirm)`<br/><br/>Include the `--full_rebuild` flag if you want to delete missing resources. |
+    | `hb preview/deploy --dbt` | Set `--dbt-root` when running `hb init`, and then use `hb build`<br/><br/>To exclude dbt models, use `hb build --no-dbt` |
```
Comment (annanixon) [inline pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx:59]: There is no longer a --no-dbt flag. Instead if you specify a filepath dbt models will not get built. So hb build = builds dbt and files hb build dashboards/* = builds only files hb build dbt:all = builds only dbt models. But you can combine multiple ex: hb build dbt:all dashboards/*

File: pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+1. Upgrade your Python package: `pip install --upgrade hashboard-cli`
+2. Run `hb init` at the root of your Hashboard project files, specifying `--dbt-root=...` if you have dbt-based models. Commit the resulting `.hbproject` file into your git repository.
+3. In your scripts / CI actions, migrate commands as follows:
+
+
+
+    | **Previous command** | **New command** |
+    | --- | --- |
+    | `hb preview (filepath)` | `hb build (filepath)` <br/><br/> Include the `--full_rebuild` flag if you want to delete missing resources. |
+    | `hb deploy (filepath)`  | If the build was already created for previewing: `hb build apply`<br/><br/>If no build has been created yet:<br/>`hb build (filepaths) && hb build apply (--no-confirm)`<br/><br/>Include the `--full_rebuild` flag if you want to delete missing resources. |
```
Comment (annanixon) [inline pages/docs/data-ops/cli/v3/v2-to-v3-migration-guide.mdx:58]: Since this was a point of confusion for Beam i'd add a new row just for hb deploy --no-preview and call out hb build && hb build apply --no-confirm as a direct replacement for clarity

File: pages/docs/data-ops/dbt-integration.mdx [hunk 8]
```diff
@@ -18,25 +18,36 @@ Hashboard can import models directly from dbt by specifying measures in your dbt
...
-      preventUpdatesFromUI: true
-      source:
-        connectionName: bigquery-prod # the name of the conneciton in Hashboard
+      connectionName: # which Hashboard data connection to use
 ```
 
 ### Define model information
 
-The simplest example of a Hashboard model just defines a measure. Hashboard models must include the `hashboard` key in the dbt model.
+The simplest example of a Hashboard model just defines a measure. Hashboard models must include the `hashboard` key in the dbt model meta.
```
Comment (annanixon) [inline pages/docs/data-ops/dbt-integration.mdx:50]: Might reword to 'Only dbt models with the `hashboard` key present in the meta field will be built by the integration' for clarity?

File: pages/docs/data-ops/dbt-integration.mdx [hunk 9]
```diff
@@ -18,25 +18,36 @@ Hashboard can import models directly from dbt by specifying measures in your dbt
...
 models:
   +meta:
     # hashboard-defaults settings will be merged into each Hashboard model
     hashboard-defaults:
       hbVersion: "1.0"
-      # preventUpdatesFromUI defaults to false, but can be set to true
-      preventUpdatesFromUI: true
-      source:
-        connectionName: bigquery-prod # the name of the conneciton in Hashboard
+      connectionName: # which Hashboard data connection to use
```
Comment (annanixon) [inline pages/docs/data-ops/dbt-integration.mdx:45]: this should be nested under a source key

File: pages/docs/data-ops/cli/v3/build.mdx
Comment (annanixon) [inline pages/docs/data-ops/cli/v3/build.mdx:1]: we should update this page to include selection syntax and the fact that you can pass multiple filepaths/glob patterns/dbt selectors

File: pages/docs/data-ops/cli/v3/pull.mdx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+# alternatively, pull by alias
+hb pull m::my_sales_model
+```
+
+## Managing dbt state with `pull`
+
+The behavior of `hb pull` is a bit different when used for dbt models. We cannot update models automatically inside of dbt files (since there could be jinja that we can't decompile).
+
+Instead of updating resources in place, we'll create new model files outside of your dbt project. This lets you inspect the configuration and merge it into your dbt files.
```
Comment (annanixon) [inline pages/docs/data-ops/cli/v3/pull.mdx:45]: We don't pull dbt by default anymore

File: pages/docs/data-ops/managing-resources-with-code.mdx (hunk 1 above)
Comment (dse0) [reply pages/docs/data-ops/managing-resources-with-code.mdx:99]: Intention here is that you can deploy the same yml files to multiple projects for templating / external data sharing. Will reword.

**Review by annanixon — APPROVED** [review]
🚀
